changelog/6447.bugfix.1.rst

```diff
@@ -0,0 +1 @@
+Fixed a bug where the private attribute ``_default_observer_coordinate`` for `~sunpy.map.GenericMap` was being used even when there was sufficient observer metadata in the header.
```

Comment:
I would maybe call this a `trivial` rather than `bugfix` entry, as it's not public facing? (And you have an entry for the public bug?)